src/main/java/io/passport/server/controller/PersonnelController.java

```diff
@@ -0,0 +1,64 @@
+package io.passport.server.controller;
+
+import io.passport.server.model.Personnel;
+import io.passport.server.repository.PersonnelRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+/**
+ * Class which stores the generated HTTP requests related to personnel operations.
+ */
+@RestController
+@RequestMapping("/personnel")
+public class PersonnelController {
+    /**
+     * Personnel repo access for database management.
+     */
+    private final PersonnelRepository personnelRepository;
+
+    @Autowired
+    public PersonnelController(PersonnelRepository personnelRepository) {
```

Comment:
I think that we can utilize @Autowired annotation directly for the property without implementing a separate method. Please give it a try and if it works, remove such redundant functions.

Reply:
Autowired annotation is observed to be causing warnings when used for the property instead of the constructor. Left as a constructor annotation for best practice.